pilot/pkg/serviceregistry/kube/controller/ambient/services.go

```diff
@@ -131,11 +140,16 @@ func (a *index) constructServiceEntries(svc *networkingclient.ServiceEntry, w *W
 	// TODO this is only checking one controller - we may be missing service vips for instances in another cluster
 	res := make([]*workloadapi.Service, 0, len(svc.Spec.Hosts))
 	for _, h := range svc.Spec.Hosts {
+		// if we have no user-provided addresses and h is not wildcarded we can try to use autoassigned addresses
+		a := addresses
+		if len(a) == 0 && !host.Name(h).IsWildCarded() {
```

Comment:
I think we need to also ignore cases where resolution==NONE? Otherwise the app is going to literally send to the allocated VIP... unless it has a route overriding that. Tricky.... need to look more at how v1 does it.

In this case, "route overriding it" would mean "has a waypoint" I suppose.

Reply:
I also added it here, probably not actually necessary since it is also in Should... per your previous comment